platform/docs/docs/deployment/docker.md

```diff
@@ -36,18 +44,198 @@ Once the Docker image has been built, it can be run as a container from the comm
+
+![OHIF in non-secure context](../assets/img/ohif-non-secure-context.png)
+
+and the following message will appear in the browser console.
+
+![browser console for non-secure context](../assets/img/browser-console-non-secure-context.png)
+
+:::info
+The above indicate that OHIF is not running in a secure context. Among other things, this means information transferred to/from OHIF is not encrypted and certain capabilities such as 3D volume loading will NOT work. However, basic stack viewport functionality will continue to function.
+
+Consideration must be given as to whether OHIF should be deployed in a secure context over SSL.
+:::
+
+### Specifying the SSL Port, Certificate and Private Key
+
+For convenience, the [built Docker image](#building-the-docker-image) can be run over SSL by
+- setting the `${SSL_PORT}` environment variable
+- volume mounting the SSL certificate
+- volume mounting the SSL private key
+
+:::info
+The volume mounted SSL certificate and private key are mapped to the [`ssl_certificate`](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_certificate) and [`ssl_certificate_key`](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_certificate_key) `nginx` directives respectively.
+:::
+
+Similar to the [`nginx` listen port](#configuring-the-nginx-listen-port), the `${SSL_PORT}` environment variable is the internal port that `nginx` listens on to serve the OHIF web server over SSL and has to be likewise published via the `-p` switch.
+
+The following is an example command running the Docker container over SSL. Note that depending on the version of Docker, an absolute path to the certificate and private key files might be required.
+
+```sh
+docker run -d -e SSL_PORT=443 -p 3003:443/tcp -v /path/to/certificate:/etc/ssl/certs/ssl-certificate.crt -v /path/to/private/key:/etc/ssl/private/ssl-private-key.key --name ohif-viewer-container ohif-viewer-image
+```
+
+:::caution
+The above deploys OHIF over SSL using `nginx`'s default SSL configuration. For further OHIF server hardening and security configuration, consider enlisting an expert and then editing OHIF's `nginx` [SSL template configuration file](https://github.com/OHIF/Viewers/blob/8a8ae237d26faf123abeb073cbf0cd426c3e9ef2/.docker/Viewer-v3.x/default.ssl.conf.template) with further [security settings](https://nginx.org/en/docs/http/ngx_http_ssl_module.html) and [tweaks](http://nginx.org/en/docs/http/configuring_https_servers.html) and then [build a new Docker image](#building-the-docker-image) from there.
+:::
+
+:::caution
+The private key is a secure entity and should have restricted access. Keep it safe!
+:::
```

Comment:
we need another caution here since i had to do chmod 644 on the generated key and crt before mounting. Apparently when you mount a volume in Docker, the permissions on the host are preserved inside the container.